odoo/addons/base/models/res_currency.py

```diff
@@ -248,26 +275,34 @@ class CurrencyRate(models.Model):
         ('currency_rate_check', 'CHECK (rate>0)', 'The currency rate must be strictly positive.'),
     ]
 
-    def _get_latest_rate(self):
-        return self.search([
-            ('currency_id', '=', self.currency_id.id),
-            ('company_id', '=', self.company_id.id or self.env.company.id),
-            ('name', '<=', self.name or fields.Date.today()),
-        ], order="name desc", limit=1)
+    def _get_latest_rate(self, excluded_name=None):
+        rates = self.currency_id.rate_ids.filtered(
+            lambda x: x.rate and x.company_id == self.company_id and x.name != excluded_name).sorted('name')
```

Comment:
You should not exclude a rate, but only consider the rates before this one.
Also the variable `excluded_name` is pointless since you can always get it from `self.name`

Reply:
changed